static/src/javascripts/projects/common/modules/video/video-container.js

```diff
@@ -0,0 +1,268 @@
+// @flow
+import bean from 'bean';
+import fastdom from 'lib/fastdom-promise';
+import $ from 'lib/$';
+import ElementInview from 'lib/element-inview';
+import videojs from 'bootstraps/enhanced/media/video-player';
+import youtube from 'common/modules/atoms/youtube';
+import detect from 'lib/detect';
+
+type State = {
+    position: number,
+    length: number,
+    videoWidth: number,
+    container: Element,
+};
+
+type Action = {
+    type: string,
+};
+
+const updateYouTubeVideo = (currentItem: ?Element) => {
+    if (currentItem != null) {
+        const youTubeAtom = currentItem.querySelector('.youtube-media-atom');
+        if (youTubeAtom) {
+            return youtube.onVideoContainerNavigation(
+                youTubeAtom.getAttribute('data-unique-atom-id')
+            );
+        }
+    }
+};
+
+const getPositionState = (position: number, length: number) => ({
+    position,
+    atStart: position === 0,
+    atEnd: position >= length,
+});
+
+const reducers = {
+    NEXT: function next(previousState: State) {
```

Comment:
You could write these as shorthand properties: `NEXT(previousState: State) { ... }`

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions